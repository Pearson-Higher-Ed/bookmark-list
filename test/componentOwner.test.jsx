/**
PEARSON PROPRIETARY AND CONFIDENTIAL INFORMATION SUBJECT TO NDA
 *  Copyright Â© 2017 Pearson Education, Inc.
 *  All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Pearson Education, Inc.  The intellectual and technical concepts contained
 * herein are proprietary to Pearson Education, Inc. and may be covered by U.S. and Foreign Patents,
 * patent applications, and are protected by trade secret or copyright law.
 * Dissemination of this information, reproduction of this material, and copying or distribution of this software
 * is strictly forbidden unless prior written permission is obtained
 * from Pearson Education, Inc.
**/
/* eslint-disable */


import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { InternationalSupport } from '@pearson-incubator/aquila-js-core';
import ComponentOwner from '../src/js/component-owner';
import msgObject from '../translations';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import { darkBlack, fullBlack } from 'material-ui/styles/colors';

 const bookmarksArr = [
  {
    "id": "1499682113569",
    "uri": "urn:pearson:manifestation:1c31193b-c5fe-4b17-881d-1f591351e82e",
    "data": {
      "baseUrl": "urn:pearson:manifestation:55a98cce-067a-4a73-b610-229959548eab"
    },
    "createdTimestamp": 1499682113569,
    "title": "Neurons and Nerves",
    "labels": "Neurons and Nerves"
  }
]
const intlObj = new InternationalSupport(msgObject, 'en');

it('renders the component correctly', () => {
  const tree = renderer
    .create(
      <MuiThemeProvider>
        <IntlProvider 
        locale={intlObj.getLocale()} 
        messages={intlObj.getMessages()}>
          <ComponentOwner 
          bookmarksArr={bookmarksArr} 
          clickBookmarkHandler={()=>{}}
          removeBookmarkHandler={()=>{}}
          ></ComponentOwner>
        </IntlProvider>
      </MuiThemeProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});




