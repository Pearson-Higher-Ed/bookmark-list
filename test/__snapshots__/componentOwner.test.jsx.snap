// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly 1`] = `
<div>
  <ul
    className="o-bookmark-list"
  >
    <li
      className="o-bookmark-section"
    >
      <a
        className="o-bookmark-content"
        data-uri="urn:pearson:manifestation:1c31193b-c5fe-4b17-881d-1f591351e82e"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        Neurons and Nerves
        <div
          className="o-bookmark-date"
        >
          <time
            value={1499682113569}
          >
            7/10/2017
          </time>
           
          <time
            value={1499682113569}
          >
            3:51 PM
          </time>
        </div>
      </a>
      <a
        aria-label="Remove Bookmarks"
        className="remove"
        onClick={[Function]}
        role="button"
        tabIndex="0"
      >
        
      </a>
    </li>
  </ul>
</div>
`;
